src/DistributedComputations/halo_communication.jl

```diff
@@ -41,33 +41,40 @@ opposite_side = Dict(
     :northeast => :southwest, 
 )
 
-# Define functions that return unique send and recv MPI tags for each side.
-# It's an integer where
-#   digit 1-2: an identifier for the field that is reset each timestep
-#   digit 3: an identifier for the field's Z-location
-#   digit 4: the side we send to/recieve from
 ID_DIGITS   = 2
 
-@inline loc_id(::Face)    = 0
-@inline loc_id(::Center)  = 1
-@inline loc_id(::Nothing) = 2
-@inline loc_id(LX, LY, LZ) = loc_id(LZ)
+location_counter = 0
+for LX in (:Face, :Center, :Nothing)
+    for LY in (:Face, :Center, :Nothing)
+        for LZ in (:Face, :Center, :Nothing)
+            @eval loc_id(::$LX, ::$LY, ::$LZ) = $location_counter
+            location_counter += 1
+        end
+    end
+end
```

Comment:
```suggestion
loc_id(lx, ly, lz) = loc_id(lx) + loc_id(ly) + loc_id(lz)
```

Reply:
But why don't you reserve a digit for each location in xyz rather than adding them together? like

```
Nothing, Nothing, Nothing = "000"
Center, Nothing, Nothing = "100"
Center, Nothing, Face = "102"
```

etc

Reply:
Oh I see, the integer has to be small

Reply:
Oh, maybe it's somethiing like

```julia
loc_id(lx, ly, lz) = loc_id(lx) + 3 * loc_id(ly) + 6 * loc_id(lz)
```

either way metaprogramming is definitely not needed here...

Reply:
Ok well then I think this is equivalent to a conversion from base-3 to base-10 so its

```
loc_id(lx, ly, lz) = loc_id(lx) + 3 * loc_id(ly) + 9 * loc_id(lz)
```

Reply:
Ah right, so the problem is equivalent to converting a number from base-3 to base-10, eg

```julia
loc_id(lx, ly, lz) = loc_id(lx) + 3 * loc_id(ly) + 9 * loc_id(lz)
```

This counts from 0 ("000") up to 26 ("222")